sim/midas/src/main/cc/emul/context.h

```diff
@@ -0,0 +1,59 @@
+#ifndef _EMUL_CONTEXT_H
+#define _EMUL_CONTEXT_H
+// clang-format off
+
+// This was copied from riscv-fesvr, which is now hosted in spike.
+// Disable formatting to minimize the diff
+
+// A replacement for ucontext.h, which is sadly deprecated.
+
+#include <pthread.h>
+
+#if defined(__GLIBC__)
+# undef USE_UCONTEXT
+# define USE_UCONTEXT
+# include <ucontext.h>
+# include <memory>
+#include <limits.h>
+
+#if (ULONG_MAX > UINT_MAX) // 64-bit systems only
+#if (100*GLIB_MAJOR_VERSION+GLIB_MINOR_VERSION < 208)
+#define GLIBC_64BIT_PTR_BUG
+static_assert (sizeof(unsigned int)  == 4, "uint size doesn't match expected 32bit");
+static_assert (sizeof(unsigned long) == 8, "ulong size doesn't match expected 64bit");
+static_assert (sizeof(void*)         == 8, "ptr size doesn't match expected 64bit");
+#endif
+#endif /* ULONG_MAX > UINT_MAX */
+
+#endif
+
+class context_t
+{
+ public:
+  context_t();
+  ~context_t();
```

Comment:
If this is not a final class, please make the destructor `virtual` otherwise it will not be called. If it is a final class, please use `class context_t final`.